scripts/testing/README.md

```diff
@@ -0,0 +1,112 @@
+
+To automatically delete the VM you need to use `--delete` option. 
+
+`./bin/test-connectors run-test my-testing-environment-name --es-version 8.12-SNAPSHOT --connectors-ref 8.12 --test-case {path to the test case file} --delete`
+
+#### Using different machine type
+All new VMs are based on a predefined image which is in turn based on `ubuntu-2204-lts` with python3 and docker installed. Cusome images are not supported. You can change a machine type by providing `--vm-type`. Visit [the official GCP documentation](https://cloud.google.com/compute/docs/general-purpose-machines) to get more information. 
+
+#### Adding a new test case
+
+A test case should be presented as a YAML file.
+
+```YAML
+---
+scenarios:
+  - index_name: search-demo-index-001
+    service_type: sharepoint_online
+    index_language: en
+    connector_configuration: scenarios/clients/spo_sean_site.json
+    native: false
+    tests:
+      - name: Full sync job is performed without errors
+        job_type: full
+        max_timeout: 60
+        match: { status: 'completed'}
+      - name: Incremental sync job is performed without errors
+        job_type: incremental
+        max_timeout: 20
+        match: { status: 'completed'}
+```
+
+Each test case can contain multiple scenarios which will be executed one by one. Each scenario can have multiple test cases which will be executed one by one too. Currently, only one type of match is supported (job status). 
+
+To add a new test case you need to create a new YAML file and describe your test case. Give it a thoughtful name. Also, you need to define `connector_configuration` that points to a connector configuration file.
+
+Example: 
+
+```JSON
+{
+  "tenant_id": "vault:path-to-spo-secret:TenantID",
+  "tenant_name": "vault:path-to-spo-secret:Tenant Name",
+  "use_text_extraction_service": false,
+  "fetch_drive_item_permissions": true,
+  "fetch_subsites": true,
+  "client_id": "vault:path-to-spo-secret:ClientID",
+  "fetch_unique_page_permissions": true,
+  "secret_value": "vault:path-to-spo-secret:SecretValue",
+  "enumerate_all_sites": false,
+  "fetch_unique_list_item_permissions": true,
+  "fetch_unique_list_permissions": true,
+  "site_collections": "CustomWebSiteCollection",
+  "use_document_level_security": false
+}
+```
+
+Make sure that all the required fields are presented in the secret. The CLI will get all the secrets and upload them to the VM.
+
+Consider adding new secrets only for new connector sources. Rich out to the Search Productivity team if you need any help with Vault secrets. 
+
+### Can I use gcloud?
+Yes, it's already installed on your machine and the cli does not limit your use. Visit [the official documentation page](https://cloud.google.com/sdk/docs) for more information.
+
+### Hygiene of use
+Since the CLI creates virtual machines in GCP it's recommended to keep them running only when you need them. Let's some trees 🌳🌳🌳. 
```

Comment:
```suggestion
Since the CLI creates virtual machines in GCP it's recommended to keep them running only when you need them. Let's save some trees 🌳🌳🌳. 
```